contrib/zpages/src/main/java/io/opencensus/zpages/TracezPageFormatter.java

```diff
@@ -0,0 +1,626 @@
+    switch (latencyBucketBoundaries) {
+      case ZERO_MICROSx10:
+        return ">0us";
+      case MICROSx10_MICROSx100:
+        return ">10us";
+      case MICROSx100_MILLIx1:
+        return ">100us";
+      case MILLIx1_MILLIx10:
+        return ">1ms";
+      case MILLIx10_MILLIx100:
+        return ">10ms";
+      case MILLIx100_SECONDx1:
+        return ">100ms";
+      case SECONDx1_SECONDx10:
+        return ">1s";
+      case SECONDx10_SECONDx100:
+        return ">10s";
+      case SECONDx100_MAX:
+        return ">100s";
+    }
+    throw new IllegalArgumentException("No value string available for: " + latencyBucketBoundaries);
+  }
+
+  private static String renderNetworkEvents(NetworkEvent networkEvent) {
+    StringBuilder stringBuilder = new StringBuilder();
+    if (networkEvent.getType() == Type.RECV) {
+      stringBuilder.append("Received");
+    } else if (networkEvent.getType() == Type.SENT) {
+      stringBuilder.append("Sent");
+    } else {
+      stringBuilder.append("Unknown");
+    }
+    stringBuilder.append(" message_id=");
+    stringBuilder.append(networkEvent.getMessageId());
+    stringBuilder.append(" message_size=");
+    stringBuilder.append(networkEvent.getMessageSize());
+    if (networkEvent.getKernelTimestamp() != null) {
+      stringBuilder.append(" kernel_timestamp=");
+      stringBuilder.append(networkEvent.getKernelTimestamp().toString());
+    }
+    return stringBuilder.toString();
+  }
+
+  private static String renderAnnotation(Annotation annotation) {
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append(annotation.getDescription());
+    if (!annotation.getAttributes().isEmpty()) {
+      stringBuilder.append(" ");
+      stringBuilder.append(renderAttributes(annotation.getAttributes()));
+    }
+    return stringBuilder.toString();
+  }
+
+  private static String renderStatus(Status status) {
+    return status.toString();
+  }
+
+  private static String renderAttributes(Map<String, AttributeValue> attributes) {
+    StringBuilder stringBuilder = new StringBuilder();
+    stringBuilder.append("Attributes:{");
+    boolean first = true;
+    for (Map.Entry<String, AttributeValue> entry : attributes.entrySet()) {
+      if (first) {
+        first = false;
+        stringBuilder.append(entry.getKey());
+        stringBuilder.append("=");
+        stringBuilder.append(attributeValueToString(entry.getValue()));
+      } else {
+        stringBuilder.append(", ");
+        stringBuilder.append(entry.getKey());
+        stringBuilder.append("=");
+        stringBuilder.append(attributeValueToString(entry.getValue()));
+      }
+    }
+    stringBuilder.append("}");
+    return stringBuilder.toString();
+  }
+
+  private static String attributeValueToString(AttributeValue attributeValue) {
+    return attributeValue.match(
+        new Function<String, String>() {
+          @Override
+          public String apply(String stringValue) {
```

Comment:
Trying to keep the code java7 compatible. I will try to investigate and understand why cannot run on openjdk7 but works with sun-jdk7.